app/services/LocationService.scala

```diff
@@ -0,0 +1,28 @@
+package services
+
+import java.sql.Timestamp
+
+import dao.LocationDao
+import javax.inject.{Inject, Singleton}
+
+import scala.concurrent.Future
+
+@Singleton
+class LocationService @Inject()(locationDao: LocationDao) {
+
+  def create(lat: Double, lon: Double, userId: Long): Future[common.Location] =
+    locationDao.create(lat, lon, userId)
+
+  def postOfflineHistory(updates: List[common.OffsetedLocationUpdate], userId: Long): Future[Option[Int]] = {
```

Comment:
Why is it called like this?

Reply:
what exactly?

Reply:
What is "offline history"?

Reply:
Location data gathered on device while it was offline and updates couldn't be sent.